### DISCUSSION TITLE
Relationship Status Overview

### DISCUSSION THREAD

**Discussion by ajbott** [PullDating/Org#164] [Ideas & Features]
## What is this?
Relationship status is a filter with special properties associated with it. Two users can enter a relationship and help lock each other’s apps to show that they’re currently in a relationship.

## What’s this for?
This feature is there to protect those in a relationship from having the partner use the app for other types of relationships and protects others that don’t want to meet someone currently in a relationship.

## How does this work?
- Notify matches looking for a relationship that matched with the user that the partner’s relationship status changed
- Apply a special border to account for users in a relationship?
- Relationship Status updated when entering a relationship
- Relationship Status locked from being changed until exiting the relationship
- Inform partner if the user tries to use the app to find someone else (hookups, another partner, etc.)
- Exiting a relationship is a one-sided consent
- Notify those in the relationship that the relationship has been ended
- Lock the account for X days before being able to start matching again (days/weeks?)
  - Gives the person time to cool down before going into another relationship
  - Prevents people from making a commitment and then moving on right away

## Overview

The Relationship Status feature introduces elements to allow users to display and manage their relationship status on the app. This feature is designed to enhance transparency and accountability among users, especially those who are currently in relationships.

### Displaying Relationship Status

- **Visibility:**
  - Users can choose to display their current relationship status on their profiles.

- **Accountability:**
  - Allows partners to know if their significant other is actively using the app for other connections.

### Entering a Relationship

- **Purpose:**
  - Protects individuals in a relationship from potential misuse of the app.

- **Notifications:**
  - Informs matches looking for a relationship about the change in the user's relationship status.

- **Visual Indicator:**
  - Applies a special border or visual element to the account to indicate the user is in a relationship.

- **Status Update:**
  - Relationship status is locked from being changed once entered.
  - Shows the relationship status prominently on the user's profile.

- **Privacy Measures:**
  - Locks the account from being found by new matches.

- **Communication:**
  - Updates other existing matches about the change.

- **Consent for Change:**
  - Requires 2-party consent to switch to a different relationship type.

### Exiting a Relationship

- **Consent Process:**
  - Allows one-sided consent for ending a relationship status.

- **Notifications:**
  - Notifies other users who are in a relationship with the individual that the relationship has ended.

- **Account Lock:**
  - Locks the account for a specified period (e.g., 1 week or 1 month) before the user can start matching again.

- **Cooldown Period:**
  - Provides a cooldown period to allow individuals time to reflect before entering into a new relationship.

- **Preventing Hasty Commitments:**
  - Prevents users from quickly moving on to the next relationship without a thoughtful cooldown period.

## Purpose

The Relationship Status feature is designed to foster transparency, accountability, and responsible use of the app, particularly for users currently in relationships. By providing clear communication and privacy measures, this feature aims to enhance the overall dating experience and promote respectful interactions.

**Comment by viktor-lesyk** [comment]
@ajbott I desperately want to highlight and comment on pieces of this text 😄 

What is a Relationship for a user? a dropdown with pre-set options? is it mandatory? 

do we also want to show who the relationship is with? what if the person B doesn't want to be shown there? isn't it too complicated?

how do we check the correctness of this info, if the input comes only from person A? 

I'd say people should be able to set any Status they want, and the options can be, for example, "Relationship" or "Open Relationship"

we can add the "Last updated" date property to show when the status was last updated/confirmed

and can send reminders, like some marketplaces do, "have you already sold this item?" or "is this item still available?"

**Reply by ajbott** [reply]
If only we lived in a world where tools were more advanced. At least we have markdown language!

1. **Definition and Flexibility of Relationship Status**:
   - Relationship status will not be mandatory. Users can opt to declare their status from a dropdown menu with options like "Single," "In a Relationship," "Open Relationship," and "Polyamorous."
   - Users have the freedom to update their status as their situation changes, reflecting the dynamic nature of relationships.

2. **Display and Privacy**:
   - Users can choose whether or not to display their relationship status publicly on their profile.
   - For relationship statuses involving multiple people (like polyamorous or open relationships), users can tag their partners only with mutual consent. If a partner chooses not to be displayed or linked, their privacy will be respected.

3. **Verification and Correctness**:
   - To ensure the authenticity of the relationship status, both parties in a two-person relationship must consent to the update. For non-monogamous relationships, the primary user can update their status, but linked partners must consent to being displayed.
   - We could certainly implement a system to periodically verify the relationship status. Similar to marketplace reminders, users will receive notifications asking if their relationship status is current, encouraging them to keep their status updated.

4. **Special Features Based on Relationship Type**:
   - **Monogamous Relationships**: Locking the account with notifications if the account is reactivated.
   - **Polyamorous Relationships**: Visibility options for multiple partners with user-controlled privacy settings.
   - **Open Relationships**: Flexibility to add new partners with existing partners being informed of changes.

5. **Data Tracking and User Support**:
   - We aim to track the duration and quality of relationships to gather insights that can help us improve user experience. For instance, measuring how long relationships last and the key factors contributing to their success.
   - We could also introduce tools for users in relationships to assess their relationship's health, offering resources like relationship counseling or advice on maintaining healthy relationships.

6. **Officializing Relationships**:
   - Making a relationship "official" on the app could involve both partners agreeing to a shared status that is then publicly displayed, enhancing transparency and accountability.
   - This feature, along with periodic verifications and mutual consent requirements, helps prevent misuse of the app and ensures a respectful environment.

**Reply by Oshyan** [reply]
I think at the very least that "Relationship Status" should be one of the basic fields users are prompted (and reminded) to fill out. And it should also be one of the very first filters built, and that filter should include being able to hide anyone who A: didn't answer or B: hides their answer. This is going to be *critical* to a large number of potential daters. A foundational desire for many people is that the person they are matching with is single! 😄 Granted we can't verify/guarantee that information, but then neither can any other dating app. The fact that someone is willing to state it openly still has value and should be encouraged, IMO.

**Comment by Oshyan** [comment]
> I desperately want to highlight and comment on pieces of this text 😄

Best that can be done is copy -> quote -> reply like this. Obviously other tools have that capability, but not free and integrated with code, issue tracking, etc. Hopefully GitHub adds it at some point... Or at least better Discourse-like quote handling.

I think in general I am a bit skeptical of such a feature. At best a dating app can encourage good behavior from its users, but certainly not enforce, and any attempt to to do is likely to backfire, not work as intended, etc. I like the idea of providing info to browsing users when they see a person's profile (e.g. last time they updated their relationship status), although there are privacy implications there for sure. I also like reminding users to update their status and be truthful/accurate, again it's encouraging good behavior, but accepting we can't enforce it.

I guess bottom line though I don't see this as necessarily its own dedicated feature, where there is some special inter-profile connection and mutual agreement, etc, etc. That feels far too proscriptive, inflexible, fiddly, etc. Some apps do have ways of connecting accounts, e.g. apps that support couples seeking others, etc., so that's an eventual (but not novel or complex) feature to consider. But I'd want to see the focus for Pull to be on the more general idea of encouraging accurate and up to date information, and signaling that to potential matches as well.

**Reply by ajbott** [reply]
### AI Enhanced Response:

Thank you for sharing your thoughts and concerns. I appreciate your perspective on the potential challenges and limitations of enforcing certain behaviors on a dating platform. 

Our intention with the Relationship Status feature isn't to enforce behavior but rather to empower users with more control over their information and how it's presented. We aim to create a flexible system that respects user privacy while also encouraging transparency and accuracy.

1. **User Control Over Information**:
   - I understand the importance of privacy and user discretion. Users should have the ability to decide exactly what information they wish to display. For instance, they can choose to show when their last relationship was, or keep such details private until they decide to share them with someone they trust.

2. **Encouraging Accurate Updates**:
   - While we can't enforce honesty, we can encourage it by making it easier and more beneficial for users to keep their information up-to-date. Regular reminders to update their status could help in maintaining the integrity of the platform, and by displaying the "last updated" date on profiles, other users can gauge the recency and reliability of the information presented.

3. **Data for Better Matches**:
   - The data collected from relationship statuses isn’t just for display. It's also intended to help us better understand what factors contribute to successful connections within our platform. By analyzing trends in relationship longevity and satisfaction, we can continually refine our matching algorithms to improve the user experience.

4. **Flexibility and Simplicity**:
   - I agree that overly complicated features could detract from user experience. Designing this feature to be as simple and unobtrusive as possible, avoiding rigid structures that might make the platform feel restrictive. The feature is optional and designed to integrate seamlessly into the user's journey, enhancing their experience without overwhelming them.

Your point about focusing on general accuracy and up-to-date information is well taken. This is precisely our goal—to build a culture of integrity and transparency that enhances every user's experience while respecting their individual privacy and preferences. This could contribute significantly to the overall effectiveness and reliability of our platform.

### Original:

>This isn't a way to do any hard enforcement. I'd say that this is more of a feature to help keep the platform up-to-date and add nice features to be able to show / hide information depending on if they've entered into a relationship. I'd like to see users have the ability to show WHAT information is displayed. If they're fine with everyone knowing when they're last relationship was, they can. If they want to keep it hidden, they can. If they want to share it with someone later, they can. They're information should be their own to control. But no information should be able to be falsified. I'd like to encourage users to use this feature to help us understand what relationships are working, and being able to track the various information that comes with the relationship status could help in that endeavor. When someone enters a relationship, how long they're in a relationship, when they exit a relationship, ect could all be valuable data points to measure the apps successfulness in offering great matches as we work to understand what creates connections.

**Reply by Oshyan** [reply]
I agree that the data would be extremely valuable, and I also would prefer the data not be falsifiable. I just don't see any good way to 1: make sure we get such data *consistently* and avoid self-selection bias (i.e. we only end up with data on relationships from people who are comfortable with/interested in sharing their relationship status with an app), and 2: enforcing that relationship status indicators are "accurate". And because of the apparent difficulty - if not impossibility - of addressing those two things I think the effort-to-reward ratio here is not balanced in favor of implementing it, at least not for a while. 

That said I'd certainly be interested in an ongoing discussion of how to address, and ideally incentivize/encourage, people to participate, be honest, etc. Part of the opportunity that an incentives-aligned dating app creates is the ability to leverage the more ethical foundation of the app and its purpose to hopefully encourage behaviors that are beneficial for the app team and the user base/community as a whole. But how exactly to do that, and how effective it could be, is unclear to me.
